src/create_disk.sh

```diff
@@ -396,11 +397,34 @@ cat > $rootfs/.coreos-aleph-version.json << EOF
 }
 EOF
 
-install_uefi() {
-    # https://github.com/coreos/fedora-coreos-tracker/issues/510
-    # See also https://github.com/ostreedev/ostree/pull/1873#issuecomment-524439883
+bootloader_install_nons390x_bootupd_epoch_1() {
+    # For background on bootupd, see https://github.com/coreos/fedora-coreos-tracker/issues/510
+    # This requires bootupd 0.2.11 in the target system, and handles both BIOS and UEFI for x86_64 for example.
+    bootupd_args=()
+    if [ "${x86_bios_bootloader}" = 1 ]; then
+        bootupd_args+=("--device=$disk")
+    fi
+    chroot_run /usr/bin/bootupctl backend install --src-root="${deploy_root}" "${bootupd_args[@]}" "${rootfs}"
+    case "${arch}" in
+        x86_64|aarch64)
+            inject_grub_uefi
+            ;;
+        ppc64le)
+            install_grub_cfg
+            ;;
+        *)
+            echo "Unhandled arch $arch"; exit 1
+    esac
+}
+
+bootupd_install_uefi_epoch_0() {
+    # For background on bootupd, see https://github.com/coreos/fedora-coreos-tracker/issues/510
     # Unshare mount ns to work around https://github.com/coreos/bootupd/issues/367
     unshare -m /usr/bin/bootupctl backend install --src-root="${deploy_root}" "${rootfs}"
+    inject_grub_uefi
```

Comment:
if we do https://github.com/coreos/coreos-assembler/pull/3631/files#r1334584042 then we'd need to add a call here:

```suggestion
    inject_grub_uefi
    inject_grub_cfg
```